make-release.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+# Release process automation script. 
+# Used to create branch/tag, update the necessary files
+# and trigger release by force pushing changes to the release branch.
+
+# set to 1 to actually trigger changes in the release branch
+TRIGGER_RELEASE=0 
+NOCOMMIT=0
+
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    '-t'|'--trigger-release') TRIGGER_RELEASE=1; NOCOMMIT=0; shift 0;;
+    '-r'|'--repo') REPO="$2"; shift 1;;
+    '-v'|'--version') VERSION="$2"; shift 1;;
+    '-n'|'--no-commit') NOCOMMIT=1; TRIGGER_RELEASE=0; shift 0;;
+    '--theia-version') THEIA_VERSION="$2"; shift 1;;
+  esac
+  shift 1
+done
+
+usage ()
+{
+  echo "Usage: $0 --repo <GIT REPO TO EDIT> --version <VERSION_TO_RELEASE> --theia-version <THEIA_VERSION> --trigger-release"
+  echo "Example: $0 --repo git@github.com:eclipse/che-theia --version 7.7.0 --theia-version 0.15.0-next.15995cd0 --trigger-release"; echo
+}
+
+if [[ ! ${VERSION} ]] || [[ ! ${REPO} ]] || [[ ! ${THEIA_VERSION} ]]; then
+  usage
+  exit 1
+fi
+
+# derive branch from version
+BRANCH=${VERSION%.*}.x
+
+# if doing a .0 release, use master; if doing a .z release, use $BRANCH
+if [[ ${VERSION} == *".0" ]]; then
+  BASEBRANCH="master"
+else 
+  BASEBRANCH="${BRANCH}"
+fi
+
+# work in tmp dir
+TMP=$(mktemp -d); pushd "$TMP" > /dev/null || exit 1
+
+# get sources from ${BASEBRANCH} branch
+echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
+git clone "${REPO}" -q
+cd "${REPO##*/}" || exit 1
+git fetch origin "${BASEBRANCH}":"${BASEBRANCH}"
+git checkout "${BASEBRANCH}"
+
+# create new branch off ${BASEBRANCH} (or check out latest commits if branch already exists), then push to origin
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+  git branch "${BRANCH}" || git checkout "${BRANCH}" && git pull origin "${BRANCH}"
+  git push origin "${BRANCH}"
+  git fetch origin "${BRANCH}:${BRANCH}"
+  git checkout "${BRANCH}"
+fi
+
+apply_files_edits () {
+  # Che Theia release may depend on Theia next or latest
+  if [[ ${THEIA_VERSION} == *"-next."* ]]; then
+    THEIA_PATCHES_DIR=master
+    THEIA_BRANCH=master
+    THEIA_GIT_REFS=refs\\\\\\\\/heads\\\\\\\\/master
+  else 
+    THEIA_PATCHES_DIR=${THEIA_VERSION}
+    THEIA_BRANCH=v${THEIA_VERSION}
+    THEIA_GIT_REFS=refs\\\\\\\\/tags\\\\\\\\/v${THEIA_VERSION}
```

Comment:
do you really need this many slashes??? 

If you run sed with the `-r` flag you probably don't have to use this many as it will use "extended regex"  and you shouldn't need to `\\` so many times. :D

Reply:
I quickly tried it but didn't work for me. So, let's keep it as is and improve later.